fetch/src/main/scala/fetch.scala

```diff
@@ -419,6 +416,15 @@ object `package` {
         )
       )
 
+    private[this] def completeOrFail[F[_], A](
+        deferred: Deferred[F, A]
+    )(value: A)(implicit F: MonadThrow[F]) =
+      deferred.complete(value).flatMap { wasSet =>
+        F.unlessA(wasSet)(
+          F.raiseError(new Exception("Attempted to complete an already-complete Deferred"))
```

Comment:
Can we specify this exception a bit more?  I don't know if creating our own exception class is worth it, but otherwise maybe a `RuntimeException`?